### PR TITLE
linux_train returns training_results_dir

### PR DESCRIPTION
**Pull request by makelinux**
Refactoring to avoid procedural programming and
rigid hard-coded key values in different files
to use functional programming and flexible style.

The "training_results" string can also be easily parameterized when it is necessary.

### CODE REVIEW

**Review by leseb — COMMENTED** [review]
Not sure why DCO does not see your sign-off, can you amend and push again? Thanks!

File: src/instructlab/train/linux_train.py [hunk 1]
```diff
@@ -396,3 +397,4 @@ def model_generate(user, **kwargs):
     model.save_pretrained("./training_results/merged_model")
 
     print("LINUX_TRAIN.PY: FINISHED")
+    return "./training_results"
```
Comment (leseb) [inline src/instructlab/train/linux_train.py:400]: Can we use a const for `training_results` since it is used and re-used in a few places?

Comment (makelinux) [reply src/instructlab/train/linux_train.py]: I've made argument with default value.

**Review by leseb — APPROVED** [review]

File: src/instructlab/train/linux_train.py [hunk 2]
```diff
@@ -393,6 +394,7 @@ def model_generate(user, **kwargs):
 
     print("LINUX_TRAIN.PY: MERGING ADAPTERS")
     model = trainer.model.merge_and_unload()
-    model.save_pretrained("./training_results/merged_model")
+    model.save_pretrained(output_dir + "/merged_model")
```
Comment (russellb) [inline src/instructlab/train/linux_train.py:397]: I don't think it matters in this case, but this is slightly preferred:

```suggestion
    model.save_pretrained(os.path.join(output_dir, "/merged_model"))
```

Comment (russellb) [reply src/instructlab/train/linux_train.py:397]: I would have applied this, but looks like maintainer access to the PR branch was turned off

**Review by leseb — APPROVED** [review]

**Review by russellb — APPROVED** [review]

**Comment by leseb** [comment]
@Mergifyio rebase

**Comment by leseb** [comment]
The branch needs to be rebased for Mergify to merge it, this is due to https://github.com/instructlab/instructlab/commit/8be639723828884dc88945b5c9572c49863ffae8. However, the lint workflow passes so I'm going ahead with this change.

**Pull request merged**